docs/source/tutorials/datalab_advanced.ipynb

```diff
@@ -0,0 +1,1241 @@
+    "}\n",
+    "\n",
+    "BINS_MAP = {\n",
+    "    \"low\": 0,\n",
+    "    \"mid\": 1,\n",
+    "    \"high\": 2,\n",
+    "}\n",
+    "\n",
+    "\n",
+    "def create_data():\n",
+    "\n",
+    "    X = np.random.rand(250, 2) * 5\n",
+    "    y = np.sum(X, axis=1)\n",
+    "    # Map y to bins based on the BINS dict\n",
+    "    y_bin = np.array([k for y_i in y for k, v in BINS.items() if v[0] <= y_i < v[1]])\n",
+    "    y_bin_idx = np.array([BINS_MAP[k] for k in y_bin])\n",
+    "\n",
+    "    # Split into train and test\n",
+    "    X_train, X_test, y_train, y_test, y_train_idx, y_test_idx = train_test_split(\n",
+    "        X, y_bin, y_bin_idx, test_size=0.5, random_state=SEED\n",
+    "    )\n",
+    "\n",
+    "    # Add several (5) out-of-distribution points. Sliding them along the decision boundaries\n",
+    "    # to make them look like they are out-of-frame\n",
+    "    X_out = np.array(\n",
+    "        [\n",
+    "            [-1.5, 3.0],\n",
+    "            [-1.75, 6.5],\n",
+    "            [1.5, 7.2],\n",
+    "            [2.5, -2.0],\n",
+    "            [5.5, 7.0],\n",
+    "        ]\n",
+    "    )\n",
+    "    # Add a near duplicate point to the last outlier, with some tiny noise added\n",
+    "    near_duplicate = X_out[-1:] + np.random.rand(1, 2) * 1e-6\n",
+    "    X_out = np.concatenate([X_out, near_duplicate])\n",
+    "\n",
+    "    y_out = np.sum(X_out, axis=1)\n",
+    "    y_out_bin = np.array([k for y_i in y_out for k, v in BINS.items() if v[0] <= y_i < v[1]])\n",
+    "    y_out_bin_idx = np.array([BINS_MAP[k] for k in y_out_bin])\n",
+    "\n",
+    "    # Add to train\n",
+    "    X_train = np.concatenate([X_train, X_out])\n",
+    "    y_train = np.concatenate([y_train, y_out])\n",
+    "    y_train_idx = np.concatenate([y_train_idx, y_out_bin_idx])\n",
+    "\n",
+    "    # Add an exact duplicate example to the training set\n",
+    "    exact_duplicate_idx = np.random.randint(0, len(X_train))\n",
+    "    X_duplicate = X_train[exact_duplicate_idx, None]\n",
+    "    y_duplicate = y_train[exact_duplicate_idx, None]\n",
+    "    y_duplicate_idx = y_train_idx[exact_duplicate_idx, None]\n",
+    "\n",
+    "    # Add to train\n",
+    "    X_train = np.concatenate([X_train, X_duplicate])\n",
+    "    y_train = np.concatenate([y_train, y_duplicate])\n",
+    "    y_train_idx = np.concatenate([y_train_idx, y_duplicate_idx])\n",
+    "\n",
+    "    py = np.bincount(y_train_idx) / float(len(y_train_idx))\n",
+    "    m = len(BINS)\n",
+    "\n",
+    "    noise_matrix = generate_noise_matrix_from_trace(\n",
+    "        m,\n",
+    "        trace=0.9 * m,\n",
+    "        py=py,\n",
+    "        valid_noise_matrix=True,\n",
+    "        seed=SEED,\n",
+    "    )\n",
+    "\n",
+    "    noisy_labels_idx = generate_noisy_labels(y_train_idx, noise_matrix)\n",
+    "\n",
+    "    # TODO: Add noise to test set when we support extra splits in Datalab\n",
```

Comment:
delete TODOs from this notebook and the other notebook as well, don't want these in tutorials